src/tests.rs

```diff
@@ -1,2 +1,229 @@
+        let res = instantiate(deps.as_mut(), env, info.clone(), msg);
+        assert!(res.is_err());
+        assert_eq!(
+            res.err().unwrap(),
+            cosmwasm_std::StdError::generic_err("Total Ownership must equal 100%."),
+        );
+    }
+
+    #[test]
+    fn test_instantiate_duplicate_owner_error() {
+        let mut deps = mock_dependencies();
+        let env = mock_env();
+        let info = mock_info("creator", &[]);
+        let msg = InstantiateMsg {
+            owners: vec![
+                Owner {
+                    address: Addr::unchecked("owner1"),
+                    ownership: 50,
+                },
+                Owner {
+                    address: Addr::unchecked("owner1"),
+                    ownership: 50,
+                },
+            ],
+        };
+
+        // Check that the `Total Ownership must equal 100%` error is thrown when the total ownership is greater than 100
+        let res = instantiate(deps.as_mut(), env, info.clone(), msg);
+        assert!(res.is_err());
+        assert_eq!(
+            res.err().unwrap(),
+            cosmwasm_std::StdError::generic_err(
+                "Duplicate owner address has been input more than once"
+            ),
+        );
+    }
+
+    #[test]
+    fn test_instantiate_0_ownership_error() {
+        let mut deps = mock_dependencies();
+        let env = mock_env();
+        let info = mock_info("creator", &[]);
+        let msg = InstantiateMsg {
+            owners: vec![
+                Owner {
+                    address: Addr::unchecked("owner1"),
+                    ownership: 100,
+                },
+                Owner {
+                    address: Addr::unchecked("owner2"),
+                    ownership: 0,
+                },
+            ],
+        };
+
+        // Check that the `Total Ownership must equal 100%` error is thrown when the total ownership is greater than 100
+        let res = instantiate(deps.as_mut(), env, info.clone(), msg);
+        assert!(res.is_err());
+        assert_eq!(
+            res.err().unwrap(),
+            cosmwasm_std::StdError::generic_err("Individual Ownership must be greater than 0."),
+        );
+    }
+
+    #[test]
+    fn test_unauthorized_error() {
+        let mut deps = mock_dependencies();
+        let env = mock_env();
+        let info = mock_info("creator", &[]);
+
+        let msg = InstantiateMsg {
+            owners: vec![
+                Owner {
+                    address: Addr::unchecked("owner1"),
+                    ownership: 45,
+                },
+                Owner {
+                    address: Addr::unchecked("owner2"),
+                    ownership: 55,
+                },
+            ],
+        };
+
+        instantiate(deps.as_mut(), env.clone(), info.clone(), msg).unwrap();
+
+        #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+        struct ContractError(pub String);
```

Comment:
What is that?

Reply:
Somehow it made it in there and stayed, but can be removed.